File: sdk/python/examples/simple_inference.py
"""
Simple inference example using MLE Python SDK
"""

import sys
import numpy as np
import mle_runtime


def main():
    if len(sys.argv) < 2:
        print("Usage: python simple_inference.py <model.mle>")
        sys.exit(1)
    
    model_path = sys.argv[1]
    
    try:
        # Create engine
        engine = mle_runtime.MLEEngine(mle_runtime.Device.CPU)
        
        # Load model
        print(f"Loading model: {model_path}")
        engine.load_model(model_path)
        
        # Print metadata
        metadata = engine.metadata
        if metadata:
            print(f"Model: {metadata.model_name}")
            print(f"Framework: {metadata.framework}")
        
        # Create input tensor (example: 1x20 features)
        input_data = np.arange(20, dtype=np.float32) * 0.1
        input_data = input_data.reshape(1, 20)
        
        # Run inference
        print("Running inference...")
        outputs = engine.run([input_data])
        
        # Print results
        print(f"Output shape: {outputs[0].shape}")
        print(f"First output value: {outputs[0][0]}")
        print(f"Peak memory usage: {engine.peak_memory_usage() / 1024:.2f} KB")
        
    except Exception as e:
        print(f"Error: {e}")
        sys.exit(1)


if __name__ == "__main__":
    main()

File: sdk/nodejs/examples/simple_inference.js
/**
 * Simple inference example using MLE Node.js SDK
 */

const { MLEEngine, Device } = require('../dist/index');

async function main() {
    if (process.argv.length < 3) {
        console.error('Usage: node simple_inference.js <model.mle>');
        process.exit(1);
    }
    
    const modelPath = process.argv[2];
    
    try {
        // Create engine
        const engine = new MLEEngine(Device.CPU);
        
        // Load model
        console.log(`Loading model: ${modelPath}`);
        await engine.loadModel(modelPath);
        
        // Print metadata
        const metadata = engine.getMetadata();
        if (metadata) {
            console.log(`Model: ${metadata.modelName}`);
            console.log(`Framework: ${metadata.framework}`);
        }
        
        // Create input tensor (example: 1x20 features)
        const input = new Float32Array(20);
        for (let i = 0; i < 20; i++) {
            input[i] = i * 0.1;
        }
        
        // Run inference
        console.log('Running inference...');
        const outputs = await engine.run([input]);
        
        // Print results
        console.log(`Output shape: [${outputs[0].length}]`);
        console.log(`First output value: ${outputs[0][0]}`);
        console.log(`Peak memory usage: ${engine.getPeakMemoryUsage() / 1024} KB`);
        
        // Clean up
        engine.dispose();
        
    } catch (error) {
        console.error(`Error: ${error.message}`);
        process.exit(1);
    }
}

main();
